### ISSUE TITLE
Unmarshalling error when estimating gas with EIP-1559 fields

### ISSUE THREAD

**Issue by dedePaulius** [web3j/web3j#1538]
# _Title_
eth_estimateGas fails with: “invalid argument 0: json: cannot unmarshal non-string into Go struct field TransactionArgs.chainId of type *hexutil.Big”

## _Description_
I was trying to implement use eth_estimateGas to estimate token transaction gas. Used constructor:

```
public Transaction(
String from,
BigInteger nonce,
BigInteger gasPrice,
BigInteger gasLimit,
String to,
BigInteger value,
String data,
Long chainId,
BigInteger maxPriorityFeePerGas,
BigInteger maxFeePerGas)
```

to construct Transaction object and pass to 
```web3j.ethEstimateGas(contractTransaction).send()```
But that results in an error ```"eth_estimateGas fails with: “invalid argument 0: json: cannot unmarshal non-string into Go struct field TransactionArgs.chainId of type *hexutil.Big"```. Checked the json body, which is being sent to the RPC, and saw, that new fields ```chainId, maxPriorityFeePerGas maxFeePerGas``` are sent in decimal form, not hex encoded. In ```Transaction``` class all integer/BigInteger field getters have such return statement: ```return convert(gas)```. So out of suspicion, tried to override the behavior of these new field getters by constructing custom ```Transaction``` class, which converts these fields to hex form, and then estimate method succeeded. Is this issue in the plans to be fixed, or could I create PR for this? I suppose, it is enough to simply add ```convert(foo)``` in these three getters?

## _Context_
web3j version: 4.8.8

**Comment by mohamedelshami** [comment]
Hi @dedePaulius

Not sure why there would be a decimal number after if the types are in BigInteger. I think however formatting to hex in the RPC request makes sense. I haven't faced that issue myself, could you please point me to where in the code you propose to make the change?

**Comment by dedePaulius** [comment]
Hi @mohamedelshami Sorry, possibly was not entirely clear. By decimal I meant base decimal, not a number having decimals. For example, if you looked at org.web3j.protocol.core.methods.request.Transaction, `gas, gasPrice, value, nonce` fields are of BigInteger type, like new ones `maxPriorityFeePerGas and maxFeePerGas`, but looking at their getters, only old ones have such form: ```convert(value)```. So I suppose it was forgotten to apply same principle to new fields, and that is the reason why estimate RPC fails. The change that is needed is just add conversion to hex using convert() method in corresponding getters ```getChainId(), getMaxPriorityFeePerGas(), getMaxFeePerGas()```. I have already tried such approach and then estimation request succeeds.

**Comment by mohamedelshami** [comment]
I see, this looks like a good fix to me since it will make the return type for these 3 getters consistent with other methods

Please submit a PR, and if possible add test coverage, and I will review it.

Thank you!

**Comment by dedePaulius** [comment]
@mohamedelshami I almost started to produce the PR, but noticed, that there is already one waiting approval: https://github.com/web3j/web3j/pull/1477 The changes implemented there look exatly the same as I wanted to implement. Can you check that PR?

**Comment by mohamedelshami** [comment]
Thanks, @dedePaulius, I've just merged the PR #1477.
